internal/util/update/common.go

```diff
@@ -0,0 +1,88 @@
+package update
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"sigs.k8s.io/kustomize/kyaml/pathutil"
+)
+
+// findAllSubpackages traverses the provided package paths
+// and finds all subpackages. A subpackage is a subdirectory underneath the
+// root that has a Kptfile in it.
+// The list is sorted in increasing order based on the depth of the subpackage
+// relative to the root package.
+func findAllSubpackages(pkgPaths ...string) ([]string, error) {
+	uniquePaths := make(map[string]bool)
+	for _, path := range pkgPaths {
+		paths, err := pathutil.DirsWithFile(path, kptfile.KptFileName, true)
+		if err != nil {
+			return []string{}, err
+		}
+		for _, p := range paths {
+			relPath, err := filepath.Rel(path, p)
+			if err != nil {
+				return []string{}, err
+			}
+			uniquePaths[relPath] = true
+		}
+	}
+	var paths []string
+	for p := range uniquePaths {
+		paths = append(paths, p)
+	}
+	sort.Slice(paths, rootPkgFirstSorter(paths))
+	return paths, nil
+}
+
+func rootPkgFirstSorter(paths []string) func(i, j int) bool {
+	return func(i, j int) bool {
+		iPath := paths[i]
+		jPath := paths[j]
+		if iPath == "." {
+			return true
+		}
+		if jPath == "." {
+			return false
+		}
+		iSegmentCount := len(strings.Split(iPath, "/"))
+		jSegmentCount := len(strings.Split(jPath, "/"))
+		return iSegmentCount < jSegmentCount
+	}
+}
+
+func subPkgFirstSorter(paths []string) func(i, j int) bool {
```

Comment:
What does this method do?

Reply:
It is used to sort package paths with `sort.Slice`. Added comments to both functions.